tmail-backend/blob/blobid-list/src/main/java/com/linagora/tmail/blob/blobid/list/SingleSaveBlobStoreDAO.java

```diff
@@ -25,17 +25,16 @@ public SingleSaveBlobStoreDAO(BlobStoreDAO blobStoreDAO,
 
     public Mono<Void> save(BucketName bucketName, BlobId blobId, byte[] data) {
         return Mono.from(blobIdList.isStored(blobId))
-                .filter(isStored -> isStored.equals(false))
-                .switchIfEmpty(Mono.empty())
-                .then(Mono.from(blobIdList.store(blobId))
-                        .then(Mono.from(blobStoreDAO.save(bucketName, blobId, data)))
-                        .onErrorResume(error -> Mono.from(blobIdList.remove(blobId)).then()))
+                .filter(isStored -> isStored.equals(true))
+                .switchIfEmpty(Mono.defer(() ->
```

Comment:
Why do we need a defer here?

Reply:
It prevents Mono call `blobStoreDAO.save(bucketName, blobId, data)` when ifNotEmpty

Reply:
We do not care, if it is empty, `blobStoreDAO.save` will not be subscribed.